### ISSUE TITLE
Fails to build with a Stack nightly

### ISSUE THREAD

**Issue by kozross** [akc/sym#1]
I've added ``sym-0.12.1`` as a dependency in a Stack project, using the ``nightly-2017-11-15`` resolver. When I try to build it, I get the following:

```
Progress: 15/16
--  While building package sym-0.12.1 using:
      /home/koz/.stack/setup-exe-cache/x86_64-linux-nopie/Cabal-simple_mPHDZzAJ_2.0.0.2_ghc-8.2.1 --builddir=.stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /home/koz/documents/programming/haskell/bdd-memetic/.stack-work/logs/sym-0.12.1.log

    Configuring sym-0.12.1...
    Preprocessing library for sym-0.12.1..
    Building library for sym-0.12.1..
    [ 1 of 20] Compiling Sym.Internal.Size ( Sym/Internal/Size.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Internal/Size.o )
    [ 2 of 20] Compiling Sym.Internal.CLongArray ( Sym/Internal/CLongArray.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Internal/CLongArray.o )
    [ 3 of 20] Compiling Sym.Internal.SubSeq ( Sym/Internal/SubSeq.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Internal/SubSeq.o )
    [ 4 of 20] Compiling Sym.Internal.Util ( Sym/Internal/Util.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Internal/Util.o )
    [ 5 of 20] Compiling Sym.Perm         ( Sym/Perm.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm.o )
    [ 6 of 20] Compiling Sym.Perm.Bijection ( Sym/Perm/Bijection.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Bijection.o )
    [ 7 of 20] Compiling Sym.Perm.D8      ( Sym/Perm/D8.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/D8.o )
    [ 8 of 20] Compiling Sym.Perm.Component ( Sym/Perm/Component.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Component.o )
    [ 9 of 20] Compiling Sym.Perm.Group   ( Sym/Perm/Group.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Group.o )
    [10 of 20] Compiling Sym.Perm.MeshPattern ( Sym/Perm/MeshPattern.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/MeshPattern.o )
    [11 of 20] Compiling Sym.Perm.Pattern ( Sym/Perm/Pattern.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Pattern.o )
    [12 of 20] Compiling Sym.Perm.SSYT    ( Sym/Perm/SSYT.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/SSYT.o )
    [13 of 20] Compiling Sym.Perm.Simple  ( Sym/Perm/Simple.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Simple.o )
    [14 of 20] Compiling Sym.Perm.Sort    ( Sym/Perm/Sort.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Sort.o )
    [15 of 20] Compiling Sym.Perm.Stat    ( Sym/Perm/Stat.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Stat.o )
    [16 of 20] Compiling Sym.Permgram     ( Sym/Permgram.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Permgram.o )
    
    /tmp/stack2264/sym-0.12.1/Sym/Permgram.hs:26:1: warning: [-Wunused-imports]
        The import of ‘Control.Applicative’ is redundant
          except perhaps to import instances from ‘Control.Applicative’
        To import instances alone, use: import Control.Applicative()
       |
    26 | import Control.Applicative
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^
    [17 of 20] Compiling Sym.Perm.Constructions ( Sym/Perm/Constructions.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Constructions.o )
    [18 of 20] Compiling Sym.Perm.Class   ( Sym/Perm/Class.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Class.o )
    [19 of 20] Compiling Sym.Perm.Meta    ( Sym/Perm/Meta.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym/Perm/Meta.o )
    [20 of 20] Compiling Sym              ( Sym.hs, .stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/build/Sym.o )
    
    /tmp/stack2264/sym-0.12.1/Sym.hs:94:5: error:
        • Constraint ‘Permutation a’ in the type of ‘unst’
            constrains only the class type variables
          Use ConstrainedClassMethods to allow it
        • When checking the class method:
            unst :: forall a. (Permutation a, Permutation a) => Perm -> a
          In the class declaration for ‘Permutation’
       |
    94 |     unst :: Permutation a => Perm -> a
       |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by kozross** [comment]
I can confirm I also get this error with the latest LTS (``lts-9.13``).

**Comment by akc** [comment]
I (hopefully) fixed this in b1642a611efbef2aa188b527e5c6c9a2612ff0cf and uploaded a new version (with some unrelated additions) to Hackage. So please try out version 0.13.0 and see if that works for you.

**Comment by kozross** [comment]
@akc Thanks for that - I can confirm that ``sym-0.13.0`` now builds under the latest nightly (``2017-11-20``) and also the latest LTS (``9.14``). Closing the issue now.
